rest-api/dao/biobank_order_dao.py

```diff
@@ -339,4 +386,119 @@ def to_client_json(self, model):
     client_json = resource.as_json()  # also validates required fields
     client_json['id'] = model.biobankOrderId
     del client_json['resourceType']
+
+    if model.orderStatus == BiobankOrderStatus.CANCELLED:
+      client_json['status'] = 'CANCELLED'
+      client_json['amendedReason'] = model.amendedReason
+      client_json['cancelledSiteId'] = model.cancelledSiteId
+      client_json['cancelledUsername'] = model.cancelledUsername
+      client_json['version'] = model.version
+
+    restored = getattr(model, 'restoredSiteId')
+    if restored:
+      client_json['status'] = 'RESTORED'
+      client_json['amendedReason'] = model.amendedReason
+      client_json['restoredSiteId'] = model.restoredSiteId
+      client_json['restoredUsername'] = model.restoredUsername
+      client_json['version'] = model.version
+
+    if model.orderStatus == BiobankOrderStatus.AMENDED:
+      client_json['amendedReason'] = model.amendedReason
+      client_json['amendedSiteId'] = model.amendedSiteId
+      client_json['amendedUsername'] = model.amendedUsername
+      client_json['version'] = model.version
+
+
     return client_json
+
+  def _do_update(self, session, order, resource):
+    order.lastModified = clock.CLOCK.now()
+    order.biobankOrderId = resource.biobankOrderId
+    order.orderStatus = BiobankOrderStatus.AMENDED
+    order.amendedTime = clock.CLOCK.now()
+    order.logPosition = LogPosition()
+    order.logPositionId = resource.logPositionId   #TODO: check this
+    # Ensure that if an order was previously cancelled/restored those columns are removed.
+    self._clear_cancelled_and_restored_fields(order)
+    self._update_history(session, order)
+    self._update_identifier_history(session, order)
+    self._update_sample_history(session, order)
+    super(BiobankOrderDao, self)._do_update(session, order, resource)
+
+  def _do_update_with_patch(self, session, order, resource):
+    order.lastModified = clock.CLOCK.now()
+    if resource['status'] == 'cancelled':
+      order.amendedReason = resource['amendedReason']
+      order.cancelledUsername = resource['cancelledInfo']['author']['value']
+      order.cancelledSiteId = get_site(resource['cancelledInfo'])
+      order.cancelledTime = clock.CLOCK.now()
+      order.orderStatus = BiobankOrderStatus.CANCELLED
+    if resource['status'] == 'restored':
+      order.amendedReason = resource['amendedReason']
+      order.restoredUsername = resource['restoredInfo']['author']['value']
+      order.restoredSiteId = get_site(resource['restoredInfo'])
+      order.restoredTime = clock.CLOCK.now()
+      order.orderStatus = BiobankOrderStatus.UNSET
```

Comment:
Is it valid to do another kind of patch? Or do we only do PATCH for changing statuses?